### PR TITLE
Replace `T::AccountId` with `<T::Lookup as StaticLookup>::Source`

### LINKED ISSUE #6042: T::AccountId used over <T::Lookup as StaticLookup>::Source in dispatchable calls
Substrate still have many `T::AccountId` in dispatchable call parameters. I think they should all be replaced by `<T::Lookup as StaticLookup>::Source`

This is results from a quick regex `\(origin, (.*): T::AccountId`

Screen Shot 2020-05-15 at 5 23 22 PM

### PR DESCRIPTION
**Pull request by Doordashcon**
Fixes #6042

polkadot address: 12zsKEDVcHpKEWb99iFt3xrTCQQXZMu477nJQsTBBrof5k2h

### CODE REVIEW

**Comment by Doordashcon** [comment]
cc @kianenigma

**Review by kianenigma — COMMENTED** [review]
Overall looks good, I wonder if anyone has an idea to reduce the clutter a bit?

**Comment by Doordashcon** [comment]
> Overall looks good, I wonder if anyone has an idea to reduce the clutter a bit?

cc @gilescope

**Comment by gilescope** [comment]
Could be a good case for using type alias? Maybe called AccoundIdLookup or some such?

**Comment by Doordashcon** [comment]
> Could be a good case for using type alias? Maybe called AccoundIdLookup or some such?

Sounds good, If this works @kianenigma I'll go right ahead and implement it.

**Comment by kianenigma** [comment]
Yeah perhaps `AccountIdLookupOf<T>`.

**Review by kianenigma — APPROVED** [review]
Happy to see this finally closed.

**Comment by kianenigma** [comment]
CI and git conflicts need to go away though.

**Comment by Doordashcon** [comment]
glad to be of help @kianenigma

**Comment by Doordashcon** [comment]
@athei please review

**Comment by xlc** [comment]
This needs label `E6-transactionversion`

**Comment by Doordashcon** [comment]
Does this PR need any more attention apart from the conflicts?

**Review by ggwpez — COMMENTED** [review]
Looks good!

Just one case where `pallet-unique::set_price` still uses `<T::Lookup as StaticLookup>::Source`.

File: frame/staking/src/benchmarking.rs [hunk 1]
```diff
@@ -83,7 +83,7 @@ pub fn create_validator_with_nominators<T: Config>(
 	let validator_prefs =
 		ValidatorPrefs { commission: Perbill::from_percent(50), ..Default::default() };
 	Staking::<T>::validate(RawOrigin::Signed(v_controller).into(), validator_prefs)?;
-	let stash_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(v_stash.clone());
+	let stash_lookup: AccountIdLookupOf<T> = T::Lookup::unlookup(v_stash.clone());
```
[the remark this answers is not in the document]
Comment (Doordashcon) [reply frame/staking/src/benchmarking.rs:86]: Noted, will resolve this

**Review by ggwpez — APPROVED** [review]

**Review by athei — APPROVED** [review]

**Comment by ggwpez** [comment]
bot merge

**Comment by Doordashcon** [comment]
Thanks for the tip!

**Pull request merged**